packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -157,6 +157,10 @@ Map<String, String> _createPubEnvironment(String context) {
   if (pubCache != null) {
     environment[_pubCacheEnvironmentKey] = pubCache;
   }
+  final String pubHostedUrl = platform.environment[_pubHostedUrlKey];
```

Comment:
This is unnecessary, since we always spawn processes with the parent environment inherited.

Reply:
Indeed we do! Removed.